### ISSUE TITLE
Missing pieces for full local SecureBoot support

### ISSUE THREAD

**Issue by cvlc12** [systemd/systemd#32801]
### Component

bootctl, kernel-install, systemd-boot

### Is your feature request related to a problem? Please describe

Nearly everything is in place to fully set up secureboot with local keys on a system, but a few steps still require extra steps or tooling

- [x] **Ukify** can generate _some_ keys (`SecureBootPrivateKey` and `SecureBootCertificate`) ~but **not** the full set of .AUTH files~ (this is done by bootcl now)
- [x] **systemd-boot** can enroll Secure Boot variables automatically (if files are available under `/loader/keys/NAME/{db,KEK,PK}.auth`)
- [x] **Ukify** can sign newly created UKIs for SecureBoot automatically
- [ ] **bootctl install/update** knows to prefer `.efi.signed` files, [but cannot](https://github.com/systemd/systemd/issues/26881) create these manually or automatically

I feel like most of the code to check all boxes already exists.

### Describe the solution you'd like

1. Define the default location for generated keys and be able to generate them all in one go
3. Teach **bootctl install/update** the `--secureboot-private-key=sb.key --secureboot-certificate=sb.cert` flags.
4. Have **Ukify** and **bootctl** automatically search the default location and sign their updates if appropriate keys are found. 

### Describe alternatives you've considered

`Ukify genkey` makes sense for PCRkeys, but might or might not be the most logical place to create the full set of SecureBoot keys.

**Comment by NekkoDroid** [comment]
>     * [ ]  **bootctl install/update** knows to prefer `.efi.signed` files, [but cannot](https://github.com/systemd/systemd/issues/26881) create these manually or automatically

This should probably be marked as done, since https://github.com/systemd/systemd/blob/main/man/systemd-sbsign.xml exists now. 

I wouldn't really expect `bootctl` to generate those files, it's job is purely to install files to some specific locations, it should be the job of what is actually generating those files to sign them.

**Comment by DaanDeMeyer** [comment]
`bootctl` now accepts `--secure-boot-auto-enroll=yes` and `systemd-sbsign` does secure boot signing so I'm pretty sure we can close this

**Comment by schildbach** [comment]
I think a missing piece is a very broad warning that this process will in many cases brick your device, e.g. Lenovo devices.

**Comment by cvlc12** [comment]
> I think we should reopen this because the behavior is even more inconsistent now:
> 
> If `systemd-bootx64.efi` hasn't been manually signed beforehand, `bootctl install --secure-boot-auto-enroll=yes` will gladly copy over an unsigned bootloader, all while it is configuring the system for secure boot...

Should I open a separate issue for this then ? because whether or not it is desirable for `bootctl install/update` to be able to sign files before installing them, the current status quo is problematic since we're installing an unsigned bootloader while configuring secure boot....

**Comment by schildbach** [comment]
Yes, currently if `/usr/lib/systemd/boot/efi/systemd-bootx64.efi` is updated (for example via distro package update) the actual bootloader in the ESP remains unchanged if there is a `/usr/lib/systemd/boot/efi/systemd-bootx64.efi.signed`.

So I agree either `bootctl install` should sign "on the fly", or mechanisms (hooks?) should be put in place for the various package managers to sign after updating the package.

(For reference, my secure boot key/cert is located at `/etc/kernel/secure-boot-certificate.pem` and `/etc/kernel/secure-boot-private-key.pem`.)
